src/geometry/types.rs

```diff
@@ -121,6 +122,89 @@ impl Vector {
     }
 }
 
+// Rectangle /////////////////////////////////////////////////////////////////
+#[derive(PartialEq, Debug)]
+pub struct Size {
+    pub width: f32,
+    pub height: f32,
+}
+
+#[derive(PartialEq, Debug)]
+pub struct Rect {
+    // the leftmost coordination system of the rectangle
+    pub origin: Point,
+    // the width and high of the rectangle
+    pub size: Size,
+}
+
+impl Rect {
+
+    pub fn half_size(&mut self) {
+        self.size.width = self.size.width/2.0;
+        self.size.height = self.size.height/2.0;
+    }
+
+    fn quarter_sized(&self) -> Rect{
+        Rect{
+            origin: Point{
+                x: self.origin.x,
+                y: self.origin.y,
+            },
+            size: Size{
```

Comment:
You should be consistent with your whitespace, usually there is a space between a word and a {